pkg/log/log.go

```diff
@@ -69,27 +67,16 @@ func (l *logger) Print(msg string) {
 			logFields[cf.Key()] = cf.Value()
 		}
 	}
-
-	l.entry = l.entry.WithFields(logFields)
-
-	switch l.level {
-	case InfoLevel:
-		l.entry.Info(msg)
-	case ErrorLevel:
-		l.entry.Error(msg)
-	case DebugLevel:
-		l.entry.Debug(msg)
-	}
+	entry := log.WithFields(logFields)
+	entry.Log(logrus.Level(l.level), msg)
 }
 
 func (l *logger) WithContext(ctx context.Context) Logger {
 	l.ctx = ctx
-	l.entry = l.entry.WithContext(ctx)
 	return l
 }
 
 func (l *logger) WithError(err error) Logger {
 	l.err = err
-	l.entry = l.entry.WithError(err)
```

Comment:
@julio-lopez how are we planning to propagate the error from here?

Reply:
Good catch. Let me fix it and add tests.

Reply:
@SupriyaKasten Addressed the issue and started adding a minimal set of tests. This particular issue is not covered by the current set of tests. I'll add more. Feel free to block the PR until enough test coverage is included.

Reply:
Thanks @julio-lopez, will block it, so that mergify does not merges it automatically before it is ready.

Reply:
@SupriyaKasten Expanded the tests, it should cover the basic functionality that is currently implemented. PTAL and LMK if I missed anything. Thanks.